recipes/setup_auth.rb

```diff
@@ -25,7 +25,7 @@
   return
 end
 
-include_recipe 'chef-splunk'
+include_recipe 'chef-splunk' if setup_auth_include?
```

Comment:
It would not fail if you setup the node data as required for the recipe to function and called the recipe directly. As it stands if you create a cookbook to wrap `chef-splunk` and don't use chef vault, you can't use this recipe to change the default splunk password and you need to add all of `setup_auth` to your wrapper.

When you get to `setup_auth` from the default recipe the data will have already been set and you shouldn't need to include it again.

Reply:
I would not recommend storing your Splunk admin credentials in node attributes. It will be readable in plain text from the node data. The chef-vault dependency is not a hard requirement; however, an encrypted data bag is a fallback. The authors of chef-vault implemented a fallback to standard encrypted data bags a while ago: 

Set
`node['chef-vault']['databag_fallback'] = true`, which is how we use this in our own splunk wrapper cookbook.

Then create a standard encrypted data bag called 'vault' -- this needs to be called vault so that it's compatible with code that assume you're using chef-vault. The data bag item is then `splunk_*` where `*` corresponds to your chef environment name.

Test Kitchen uses a data bag that isn't encrypted with the chef-vault fallback flag set to true.

```
% cat data_bags/vault/splunk__default.json
{
  "id": "splunk__default",
  "auth": "admin:notarealpassword",
  "secret": "notarealsecret"
}
```
I see an opportunity to update the README to clarify this point, though.

Reply:
the chef-vault cookbook provides the `chef_vault_item` method that has built-in logic for the fallback to encrypted data bag and even to use non-encrypted data bags (for testing).

https://github.com/chef-cookbooks/chef-splunk/blob/master/recipes/default.rb#L22-L23

Reply:
https://github.com/chef-cookbooks/chef-vault/blob/master/libraries/helpers.rb#L36-L44

Reply:
While storing a cluster admin passwords in Chef node data would be very bad, we are trying to just change the default password for the universal forwarder.

We have a functional workaround, but being able to pass the username / password into the `setup_auth` recipe from this cookbook would be and being able to skip the recipe looping back to `chef-splunk::default` would be nicer.

The other option would be to delete the line, but that seems like a breaking change if someone is calling `chef-splunk::client` directly.

If you don't think the approach. I can cheery pick the sysV fix into a new PR.

Reply:
Thanks we had missed that fall back. I'll raise a new PR.

Reply:
I can see where an enhancement would work is to check whether `node.run_state['splunk_auth_info']` has already been set, then it skips that portion of the default recipe.

Reply:
I think using the databag and the setting the `node['chef-vault']['databag_fallback'] = true` is a way better solution. To what we were doing. Totally kicking ourselves that we missed that, tomorrow when we have updated our cookbook either @JonoRicci or myself will raise a PR to update the README.

Thanks for your help on this one. The single line fix for centos-6 and amazon linux 1 is now in a new PR.